components/needs_team/needs_team.jsx

```diff
@@ -84,6 +84,12 @@ export default class NeedsTeam extends React.Component {
             finishedFetchingChannels: false,
         };
 
+        // When the user comes from mobile view redirected from get mobile app
+        // screen, redirect him to /
+        if (!team && this.props.match.params.team === 'switch_team') {
```

Comment:
This is odd. Why do we route to `/:team/select_team` in the first place?. Shouldn't it be just `/select-team`?. that way we don't even enter this route.

Reply:
I think the url in mobile screens should be  to be changed to `/select_team` instead if `switch_team` as well

Reply:
I don't think so. The select_team will show you the team selector, with switch_team you go to the "default" channel, that was the previous behavior. We can use other approach, but being a fix for the release, I prefer to keep it as simple as possible.

Reply:
I am with Sudheer here. This doesn't make sense to me. 'switch_team' is a perfectly valid team name. Shouldn't we fix whatever is redirecting to the wrong place instead of patching this here?

Reply:
https://github.com/mattermost/mattermost-webapp/blob/4af1ff27aea3f0bc8a90577c1d7e021b7f585347/components/get_ios_app/get_ios_app.jsx#L55

I think it is related to this line here.  We might change this instead as there is not `/switch_team` route.


    <Link to='/'>
        <FormattedMessage
            id='get_app.continueWithBrowserLink'
            defaultMessage='continue with browser'
        />
    </Link>  

so we redirect to the default team.

Reply:
Sure, and we can, but I prefer to leave the fix as simple as possible as far as is for release-5.1. Anyway, I can rethink the PR and see if the proper solution is simple too.

Reply:
@sudheerDev Yes, I think this solution is simpler and better. I though there was some kind of check in "/" to show the "get mobile app" page, and actually, it is, but it store in the local store if you already have seen it.